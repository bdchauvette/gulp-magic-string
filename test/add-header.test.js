const { addHeader } = require("..")
const { fibonacci } = require("./fixtures")
const { transform } = require("./helpers")

describe("string header", () => {
  it("adds the provided string to the end of the file contents", async () => {
    const file = fibonacci.toVinyl()
    const transformedFile = await transform(file, addHeader("ğŸ—£ğŸ—£ğŸ—£"))
    expect(transformedFile.contents.toString()).toMatchSnapshot()
  })
})

describe("function header", () => {
  it("passes the file to the function", async () => {
    const file = fibonacci.toVinyl()
    const header = jest.fn()
    await transform(file, addHeader(header))
    expect(header).toHaveBeenCalledTimes(1)
    expect(header).toHaveBeenLastCalledWith(file)
  })

  it("adds the generated header to the end of the file contents", async () => {
    const file = fibonacci.toVinyl()
    const header = (f) => `ğŸ—£ğŸ—£ğŸ—£ - ${f.basename}`
    const transformedFile = await transform(file, addHeader(header))
    expect(transformedFile.contents.toString()).toMatchSnapshot()
  })
})
