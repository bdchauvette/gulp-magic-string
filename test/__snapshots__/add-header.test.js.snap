// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function header adds the generated header to the end of the file contents 1`] = `
"ðŸ—£ðŸ—£ðŸ—£ - fibonacci.js
export function fibonacci(num) {
  if (num > 1) {
    return fibonacci(num - 1) + fibonacci(num - 2)
  }

  return 1
}
"
`;

exports[`string header adds the provided string to the end of the file contents 1`] = `
"ðŸ—£ðŸ—£ðŸ—£
export function fibonacci(num) {
  if (num > 1) {
    return fibonacci(num - 1) + fibonacci(num - 2)
  }

  return 1
}
"
`;
